localstack-core/localstack/services/stepfunctions/asl/component/intrinsic/argument/argument.py

```diff
@@ -0,0 +1,92 @@
+import abc
+from typing import Any, Final, Optional
+
+from localstack.services.stepfunctions.asl.component.common.string.string_expression import (
+    StringVariableSample,
+)
+from localstack.services.stepfunctions.asl.component.eval_component import EvalComponent
+from localstack.services.stepfunctions.asl.eval.environment import Environment
+from localstack.services.stepfunctions.asl.utils.json_path import extract_json
+
+
+class Argument(EvalComponent, abc.ABC):
```

Comment:
doc: A docstring explaining the motivation behind the argument abstraction would be very helpful here.